kartothek/io/testing/update.py

```diff
@@ -662,7 +662,14 @@ def test_update_first_time_with_secondary_indices(
     )
 
 
-def test_partition_on_null(store_factory, bound_update_dataset):  # gh-262
+@pytest.fixture
+def expected_error():
```

Comment:
Please add a comment here, to explain the rationale of this logic

Reply:
When I call metaparttion.partition_on using yamal_pipeline, it throws YamalError. 
test_partition_on_null  need to have a common fixture to support various testing backends.